scripts/zones/Inner_Horutoto_Ruins/mobs/Goblin_Weaver.lua

```diff
@@ -2,10 +2,32 @@
 -- Area: Inner Horutoto Ruins
 --  Mob: Goblin Weaver
 -----------------------------------
+local ID = zones[xi.zone.INNER_HORUTOTO_RUINS]
+-----------------------------------
 local entity = {}
+local nmPopChance = 50
 
 entity.onMobDeath = function(mob, player, optParams)
     xi.regime.checkRegime(player, mob, 648, 2, xi.regime.type.GROUNDS)
 end
 
+entity.onMobDespawn = function(mob)
+    local totd = VanadielTOTD()
+    if totd == xi.time.NIGHT or totd == xi.time.MIDNIGHT then
+        if math.random(1, 100) <= nmPopChance then
+            local mobId = mob:getID()
+            local nmId = ID.mob.MAGICKED_BONES_PH_DAGGER[mobId]
+            if nmId then
+                local skele = GetMobByID(nmId)
+                local respawnTime = mob:getRespawnTime()
+                DisallowRespawn(mobId, true)
+                mob:timer(respawnTime, function()
+                    skele:setModelId(1096) --Dagger Skeleton Model ID
+                    skele:spawn()
```

Comment:
I assume they are supposed to spawn immediately upon killing the goblin? otherwise you could use the phMobDespawn function. It doesn't work of the PH has time-/weather-based spawn conditions, but should work fine in this direction

Reply:
Actually that function supports immediate spawn of the NM as well: 

`xi.mob.phOnDespawn(mob, ID.mob.MAGICKED_BONES_PH_DAGGER, 50, <delayTilNextNMWindow>, true)`

Reply:
The timer was to make them spawn after a delay, otherwise yes they do spawn immediately. I wasn't sure which behavior was the desired and/or appropriate one, so I went with it spawning after a normal respawn length

Reply:
sidenote: comments are `-- ` in this project not `--` you need the space.

Reply:
Ah alright, I can fix that easily enough. I guess I missed that when looking at the style guidelines.

Reply:
> so I went with it spawning after a normal respawn length

Ah, definitely use the ph helper function then. In the magicked_bones lua that you make to fix their model ids, you can put in `onMobSpawn` to despawn these particular ones immediately if they're outside of the valid ToTD